orttraining/orttraining/python/training/ortmodule/_execution_agent.py

```diff
@@ -2,134 +2,327 @@
-                 providers=None, provider_options=None):
-        """
-        :param path_or_bytes: filename or serialized ONNX or ORT format model in a byte string
-        :param fw_feed_names: Feed names for foward pass.
-        :param fw_outputs_device_info: Device info for fetches in forward pass.
-        :param bw_fetches_names: Fetch names for backward pass.
-        :param bw_outputs_device_info: Device info for fetches in backward pass.
-        :param sess_options: session options
-        :param providers: Optional sequence of providers in order of decreasing
-            precedence. Values can either be provider names or tuples of
-            (provider name, options dict). If not provided, then all available
-            providers are used with the default precedence.
-        :param provider_options: Optional sequence of options dicts corresponding
-            to the providers listed in 'providers'.
-
-        The model type will be inferred unless explicitly set in the SessionOptions.
-        To explicitly set:
-          so = onnxruntime.SessionOptions()
-          so.add_session_config_entry('session.load_model_format', 'ONNX') or
-          so.add_session_config_entry('session.load_model_format', 'ORT') or
-
-        A file extension of '.ort' will be inferred as an ORT format model.
-        All other filenames are assumed to be ONNX format models.
+    def __init__(self, onnx_model: ModelProto, device: torch.device, session_options: Optional[SessionOptions] = None,
+                 providers: Optional[List[Union[str, Tuple[str, Dict]]]] = None, provider_options: Optional[List[Dict]] = None):
+        """Initializes TrainingAgent
+
+        Args:
+            onnx_model: ONNX ModelProto object to be wrapped
+            device: torch device where the computation should happen
+            sess_options: session options
+            providers: Optional sequence of providers in order of decreasing
+                precedence. Values can either be provider names or tuples of
+                (provider name, options dict). If not provided, then all available
+                providers are used with the default precedence.
+            provider_options: Optional sequence of options dicts corresponding
+                to the providers listed in 'providers'.
 
         'providers' can contain either names or names and options. When any options
         are given in 'providers', 'provider_options' should not be used.
 
         The list of providers is ordered by precedence. For example ['CUDAExecutionProvider', 'CPUExecutionProvider']
         means execute a node using CUDAExecutionProvider if capable, otherwise execute using CPUExecutionProvider.
         """
-
-        self._inference_session = onnxruntime.InferenceSession(path_or_bytes, session_options,
+        super().__init__(onnx_model, device)
+        self._info = YieldOpInfo.from_training_model(onnx_model)
```

Comment:
```suggestion
        self._yield_op_info = YieldOpInfo.from_training_model(onnx_model)
```
_info is too vague :)